core/src/main/java/com/linecorp/armeria/client/retry/RetryConfig.java

```diff
@@ -0,0 +1,163 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.Response;
+
+/**
+ * Holds retry config used by a {@link RetryingClient}.
+ * A {@link RetryConfig} instance encapsulates the used {@link RetryRule}, maxTotalAttempts,
+ *  * and responseTimeoutMillisForEachAttempt.
+ */
+public final class RetryConfig<T extends Response> {
+    private final int maxTotalAttempts;
+    private final long responseTimeoutMillisForEachAttempt;
+    @Nullable private final RetryRule retryRule;
+    @Nullable private final RetryRuleWithContent<T> retryRuleWithContent;
+    @Nullable private final RetryRule fromRetryRuleWithContent;
+    private final int maxContentLength;
+    private final boolean needsContentInRule;
+
+    /**
+     * Returns a new {@link RetryConfigBuilder} with the default values from Flags.
+     * Uses a {@link RetryRule}.
+     */
+    public static <T extends Response> RetryConfigBuilder<T> builder(RetryRule retryRule) {
+        return new RetryConfigBuilder<>(retryRule);
+    }
+
+    /**
+     * Returns a new {@link RetryConfigBuilder} with the default values from Flags.
+     * Uses a {@link RetryRuleWithContent} with unlimited content length.
+     */
+    public static <T extends Response> RetryConfigBuilder<T> builder(
+            RetryRuleWithContent<T> retryRuleWithContent) {
+        return new RetryConfigBuilder<>(retryRuleWithContent);
+    }
+
+    /**
+     * Returns a new {@link RetryConfigBuilder} with the default values from Flags.
+     * Uses {@link RetryRuleWithContent} with specified maxContentLength.
+     */
+    public static <T extends Response> RetryConfigBuilder<T> builder(
+            RetryRuleWithContent<T> retryRuleWithContent, int maxContentLength) {
+        return new RetryConfigBuilder<>(retryRuleWithContent, maxContentLength);
+    }
+
+    /**
+     * Returns config's maxTotalAttempt.
+     */
+    public int maxTotalAttempts() {
+        return maxTotalAttempts;
+    }
+
+    /**
+     * Returns config's responseTimeoutMillisForEachAttempt.
+     */
+    public long responseTimeoutMillisForEachAttempt() {
+        return responseTimeoutMillisForEachAttempt;
+    }
+
+    /**
+     * Returns config's retryRule, could be null.
+     */
+    @Nullable public RetryRule retryRule() {
```

Comment:
Could you add a new line at the end of `@Nullable`?
```suggestion
    @Nullable
    public RetryRule retryRule() {
```

Reply:
Done